crates/control/src/motion/joint_command_sender.rs

```diff
@@ -1,83 +1,96 @@
+        let head_joints_command = context.head_joints_command;
+        let motion_selection = context.motion_selection;
+        let arms_up_squat = context.arms_up_squat_joints_command;
+        let jump_left = context.jump_left_joints_command;
+        let jump_right = context.jump_right_joints_command;
+        let sit_down = context.sit_down_joints_command;
+        let stand_up_back_positions = context.stand_up_back_positions;
+        let stand_up_front_positions = context.stand_up_front_positions;
+        let walk = context.walk_joints_command;
+
+        let (positions, stiffnesses) = match motion_selection.current_motion {
+            MotionType::ArmsUpSquat => (arms_up_squat.positions, arms_up_squat.stiffnesses),
+            MotionType::Dispatching => (
+                dispatching_command.positions,
+                dispatching_command.stiffnesses,
+            ),
+            MotionType::FallProtection => (fall_protection_positions, fall_protection_stiffnesses),
+            MotionType::JumpLeft => (jump_left.positions, jump_left.stiffnesses),
+            MotionType::JumpRight => (jump_right.positions, jump_right.stiffnesses),
+            MotionType::Penalized => (*context.penalized_pose, Joints::fill(0.8)),
+            MotionType::SitDown => (sit_down.positions, sit_down.stiffnesses),
+            MotionType::Stand => (
+                Joints::from_head_and_body(head_joints_command.positions, walk.positions),
+                Joints::from_head_and_body(head_joints_command.stiffnesses, walk.stiffnesses),
+            ),
+            MotionType::StandUpBack => (*stand_up_back_positions, Joints::fill(1.0)),
+            MotionType::StandUpFront => (*stand_up_front_positions, Joints::fill(1.0)),
+            MotionType::Unstiff => (current_positions, Joints::fill(0.0)),
+            MotionType::Walk => (
+                Joints::from_head_and_body(head_joints_command.positions, walk.positions),
+                Joints::from_head_and_body(head_joints_command.stiffnesses, walk.stiffnesses),
+            ),
+        };
         context
             .hardware_interface
-            .write_to_actuators(
-                Joints::default(),
-                Joints::default(),
-                Leds {
-                    left_ear: 0.0.into(),
-                    right_ear: 0.0.into(),
-                    chest: self.last_average_color,
-                    left_foot: self.last_average_color,
-                    right_foot: self.last_average_color,
-                    left_eye: self.last_average_color.into(),
-                    right_eye: self.last_average_color.into(),
-                },
-            )
+            .write_to_actuators(positions, stiffnesses, context.leds.clone())
             .wrap_err("failed to write to actuators")?;
-        Ok(MainOutputs::default())
+
+        Ok(MainOutputs {
+            positions: positions.into(),
+            stiffnesses: stiffnesses.into(),
```

Comment:
Why do we output the positions and stiffnesses here? Is there any node that uses that? Otherwise these should be additional outputs, right?

Reply:
That might be true in the new framework. Previously the cycler was responsible of hardware communication. I removed the outputs and moved them to additional outputs